LexicalPreservingPrinter: Bug or request for adding Annotations to MethodDeclaration

Unfortunately I am having a hard time to get anything to work with annotations.

Using this simple File as input
```
public class TestPage extends Page {

   protected void test() {}

   @Override
   protected void initializePage() {}
}
```

and this code for the printing and changing of the CompilationUnit
```
		try (BufferedReader reader = Files.newBufferedReader(filePath)) {

			Pair<ParseResult<CompilationUnit>, LexicalPreservingPrinter> result = LexicalPreservingPrinter
					.setup(ParseStart.COMPILATION_UNIT, Providers.provider(reader));

			CompilationUnit cu = result.a.getResult().get();

			changeCU(cu);
		} catch (IOException e) {
			e.printStackTrace();
		}

	private void changeCU(CompilationUnit cu) {
		cu.getTypes().stream()
			.forEach(type -> {
				type.getMembers().stream()
					.forEach(member -> {
						if (member instanceof MethodDeclaration) {
							MethodDeclaration methodDeclaration = (MethodDeclaration) member;
							methodDeclaration.addAnnotation("Override");
						}
					});
			});
	}
```

Running this example I always get an UnsupportedOperationException.
Here is the important part of the stack trace
```
Caused by: java.lang.UnsupportedOperationException: Override
	at com.github.javaparser.printer.lexicalpreservation.LexicalDifferenceCalculator.calculatedSyntaxModelForNode(LexicalDifferenceCalculator.java:212)
	at com.github.javaparser.printer.lexicalpreservation.LexicalDifferenceCalculator.lambda$calculatedSyntaxModelForNode$0(LexicalDifferenceCalculator.java:134)
	at java.util.Arrays$ArrayList.forEach(Arrays.java:3880)
	at com.github.javaparser.printer.lexicalpreservation.LexicalDifferenceCalculator.calculatedSyntaxModelForNode(LexicalDifferenceCalculator.java:134)
	at com.github.javaparser.printer.lexicalpreservation.LexicalDifferenceCalculator.calculatedSyntaxModelForNode(LexicalDifferenceCalculator.java:123)
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter.interpret(LexicalPreservingPrinter.java:308)
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter.prettyPrintingTextNode(LexicalPreservingPrinter.java:304)
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter.getOrCreateNodeText(LexicalPreservingPrinter.java:325)
	at com.github.javaparser.printer.lexicalpreservation.ChildTextElement.getNodeTextForWrappedNode(ChildTextElement.java:57)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.reverseIterator(TextElementIteratorsFactory.java:170)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.partialReverseIterator(TextElementIteratorsFactory.java:184)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.reverseIterator(TextElementIteratorsFactory.java:178)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.reverseIterator(TextElementIteratorsFactory.java:171)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.partialReverseIterator(TextElementIteratorsFactory.java:184)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.reverseIterator(TextElementIteratorsFactory.java:178)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.reverseIterator(TextElementIteratorsFactory.java:171)
	at com.github.javaparser.printer.lexicalpreservation.TextElementIteratorsFactory.partialReverseIterator(TextElementIteratorsFactory.java:184)
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter.tokensPreceeding(LexicalPreservingPrinter.java:248)
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter.findIndentation(LexicalPreservingPrinter.java:333)
	at com.github.javaparser.printer.lexicalpreservation.Difference.apply(Difference.java:379)
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter$1.concreteListChange(LexicalPreservingPrinter.java:167)
	at com.github.javaparser.ast.observer.PropagatingAstObserver.listChange(PropagatingAstObserver.java:72)
	at com.github.javaparser.ast.NodeList.lambda$notifyElementAdded$1(NodeList.java:409)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.github.javaparser.ast.NodeList.notifyElementAdded(NodeList.java:409)
	at com.github.javaparser.ast.NodeList.add(NodeList.java:67)
	at com.github.javaparser.ast.nodeTypes.NodeWithAnnotations.addAnnotation(NodeWithAnnotations.java:71)
```

Is there a reason why this is not supported? Or is it maybe a bug?

Kind regards

--- a/javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java
+++ b/javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java
@@ -209,7 +209,7 @@ class LexicalDifferenceCalculator {
         } else if (csm instanceof CsmAttribute) {
             CsmAttribute csmAttribute = (CsmAttribute)csm;
             Object value = change.getValue(csmAttribute.getProperty(), node);
-            throw new UnsupportedOperationException((String)value);
+            elements.add(new CsmToken(csmAttribute.getTokenType(value.toString()), value.toString()));
         } else {
             throw new UnsupportedOperationException(csm.getClass().getSimpleName());
         }

--- a/javaparser-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinterTest.java
+++ b/javaparser-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinterTest.java
@@ -511,19 +511,6 @@ public class LexicalPreservingPrinterTest extends AbstractLexicalPreservingTest 
         assertEquals(code, lpp.print(cu));
     }
 
-    private void changeCU(CompilationUnit cu) {
-        cu.getTypes().stream()
-                .forEach(type -> {
-                    type.getMembers().stream()
-                            .forEach(member -> {
-                                if (member instanceof MethodDeclaration) {
-                                    MethodDeclaration methodDeclaration = (MethodDeclaration) member;
-                                    methodDeclaration.addAnnotation("Override");
-                                }
-                            });
-                });
-    }
-
     // See issue #855
     @Test
     public void handleOverrideAnnotation() {
@@ -540,7 +527,26 @@ public class LexicalPreservingPrinterTest extends AbstractLexicalPreservingTest 
 
         CompilationUnit cu = result.a.getResult().get();
 
-        changeCU(cu);
+        cu.getTypes().stream()
+                .forEach(type -> {
+                    type.getMembers().stream()
+                            .forEach(member -> {
+                                if (member instanceof MethodDeclaration) {
+                                    MethodDeclaration methodDeclaration = (MethodDeclaration) member;
+                                    if (!methodDeclaration.getAnnotationByName("Override").isPresent()) {
+                                        methodDeclaration.addAnnotation("Override");
+                                    }
+                                }
+                            });
+                });
+        assertEquals("public class TestPage extends Page {\n" +
+                "\n" +
+                "   @Override()\n" +
+                "   protected void test() {}\n" +
+                "\n" +
+                "   @Override\n" +
+                "   protected void initializePage() {}\n" +
+                "}", result.b.print(cu));
     }
 
 }

--- a/javaparser-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinterTest.java
+++ b/javaparser-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinterTest.java
@@ -1,10 +1,10 @@
 package com.github.javaparser.printer.lexicalpreservation;
 
 import com.github.javaparser.GeneratedJavaParserConstants;
-import com.github.javaparser.ast.ArrayCreationLevel;
-import com.github.javaparser.ast.ImportDeclaration;
-import com.github.javaparser.ast.Node;
-import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ParseStart;
+import com.github.javaparser.Providers;
+import com.github.javaparser.ast.*;
 import com.github.javaparser.ast.body.*;
 import com.github.javaparser.ast.expr.ArrayCreationExpr;
 import com.github.javaparser.ast.expr.BinaryExpr;
@@ -15,9 +15,12 @@ import com.github.javaparser.ast.stmt.Statement;
 import com.github.javaparser.ast.stmt.TryStmt;
 import com.github.javaparser.ast.type.Type;
 import com.github.javaparser.ast.type.UnionType;
+import com.github.javaparser.utils.Pair;
 import org.junit.Test;
 
+import java.io.BufferedReader;
 import java.io.IOException;
+import java.nio.file.Files;
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
@@ -506,6 +509,38 @@ public class LexicalPreservingPrinterTest extends AbstractLexicalPreservingTest 
         considerCode(code);
 
         assertEquals(code, lpp.print(cu));
+    }
+
+    private void changeCU(CompilationUnit cu) {
+        cu.getTypes().stream()
+                .forEach(type -> {
+                    type.getMembers().stream()
+                            .forEach(member -> {
+                                if (member instanceof MethodDeclaration) {
+                                    MethodDeclaration methodDeclaration = (MethodDeclaration) member;
+                                    methodDeclaration.addAnnotation("Override");
+                                }
+                            });
+                });
+    }
+
+    // See issue #855
+    @Test
+    public void handleOverrideAnnotation() {
+        String code = "public class TestPage extends Page {" + EOL +
+                EOL +
+                "   protected void test() {}" + EOL +
+                EOL +
+                "   @Override" + EOL +
+                "   protected void initializePage() {}" + EOL +
+                "}";
+
+        Pair<ParseResult<CompilationUnit>, LexicalPreservingPrinter> result = LexicalPreservingPrinter
+                .setup(ParseStart.COMPILATION_UNIT, Providers.provider(code));
+
+        CompilationUnit cu = result.a.getResult().get();
+
+        changeCU(cu);
     }
 
 }

